### PR TITLE
[PSDK-86] Enable creation of wallets backed by server signers

### PR DESCRIPTION
**Pull request by alex-stone**
### What changed? Why?
This enables developers to create wallets that are backed by server signers v.s. managing the seeds themselves.

When calling `user.create_wallet(server_signer: true)` the wallet will be created and associated with the project's default server signer (if one exists). It will return an error if a server signer does not exist for that project.

### Default address creation

Rather than add further cases to the wallet initialization this removes the side-effects from that method in favor of a `Wallet.create` class method.

This code path will only be used for creation of wallets, where we will additionally create a default address for wallets that are not backed by server signers.

### Network IDs

This also adds support for specifying the `network_id` which the backend will properly validate based on supported networks.

#### Qualified Impact

This touches the `user.create_wallet` flow

### CODE REVIEW

File: lib/coinbase/wallet.rb [hunk 1]
```diff
@@ -35,6 +35,33 @@ def import(data)
...
+      # @param network_id [String] (Optional) the ID of the blockchain network. Defaults to 'base-sepolia'.
+      # @param server_signer [Boolean] (Optional) whether Wallet should use project's server signer. Defaults to false.
+      # @return [Coinbase::Wallet] the new Wallet
+      def create(network_id: 'base-sepolia', server_signer: false)
+        model = Coinbase.call_api do
+          wallets_api.create_wallet(
+            create_wallet_request: {
+              wallet: {
+                network_id: network_id,
+                use_server_signer: server_signer
```
Comment (jazz-cb) [inline lib/coinbase/wallet.rb:49]: could you add this to the readme as well?

**Review by jazz-cb — APPROVED** [review]

**Review by jazz-cb — APPROVED** [review]

File: lib/coinbase/wallet.rb (hunk 1 above)
Comment (alex-stone) [reply lib/coinbase/wallet.rb:49]: Synced offline, we will update the readme once external developers are able to create and run server signers on their own.

File: lib/coinbase/user.rb [hunk 2]
```diff
@@ -20,21 +20,11 @@ def id
     end
 
     # Creates a new Wallet belonging to the User.
+    # @param network_id [String] (Optional) the ID of the blockchain network. Defaults to 'base-sepolia'.
+    # @param server_signer [Boolean] (Optional) whether Wallet should use project's server signer. Defaults to false.
```
Comment (yuga-cb) [inline lib/coinbase/user.rb:24]: Should we document saying this is a no-op right now?

File: lib/coinbase/wallet.rb (hunk 1 above)
Comment (yuga-cb) [reply lib/coinbase/wallet.rb:49]: +1 to keeping it semi-hidden for now :)

File: lib/coinbase/user.rb (hunk 2 above)
Comment (alex-stone) [reply lib/coinbase/user.rb:24]: Moving to a feature branch!

**Comment by alex-stone** [comment]
@jazz-cb @yuga-cb 

### Changes
* Moved to a feature branch
* We'll land this to the next candidate version once we support server signer wallet seed creation + address generation from the SDK

**Pull request merged**